apps/backend/src/modules/quiz-answers/quiz-answer.repository.ts

```diff
@@ -0,0 +1,108 @@
+import { DatabaseTableName } from "~/libs/modules/database/database.js";
+import { type Repository } from "~/libs/types/types.js";
+
+import {
+	type CategoriezedQuizAnswerModel,
+	type UserAnswer,
+} from "./libs/types/types.js";
+import { QuizAnswerEntity } from "./quiz-answer.entity.js";
+import { type QuizAnswerModel } from "./quiz-answer.model.js";
+
+class QuizAnswerRepository implements Repository {
+	private quizAnswerModel: typeof QuizAnswerModel;
+
+	public constructor(quizAnswerModel: typeof QuizAnswerModel) {
+		this.quizAnswerModel = quizAnswerModel;
+	}
+
+	public create(): Promise<null> {
+		return Promise.resolve(null);
+	}
+
+	public async createUserAnswers(
+		userId: number,
+		answerIds: number[],
+	): Promise<UserAnswer[]> {
+		const answers = await Promise.all(
+			answerIds.map((answerId) =>
+				this.quizAnswerModel
+					.query()
+					.from(DatabaseTableName.QUIZ_ANSWERS_TO_USERS)
+					.insertAndFetch({
+						answerId,
+						userId,
+					}),
+			),
+		);
+
+		return answers.map((answer) => ({
+			answerId: answer.answerId,
+			createdAt: answer.createdAt,
+			id: answer.id,
+			updatedAt: answer.updatedAt,
+			userId: answer.userId,
+		}));
+	}
+
+	public delete(): ReturnType<Repository["delete"]> {
+		return Promise.resolve(true);
+	}
+
+	public async deleteUserAnswers(userId: number): Promise<number> {
+		return await this.quizAnswerModel
+			.query()
+			.from(DatabaseTableName.QUIZ_ANSWERS_TO_USERS)
+			.where("userId", userId)
+			.delete();
+	}
+
+	public find(): Promise<null> {
+		return Promise.resolve(null);
+	}
+
+	public findAll(): Promise<null[]> {
+		return Promise.resolve([null]);
+	}
+
+	public async findByIds(ids: number[]): Promise<QuizAnswerEntity[]> {
+		const items = await this.quizAnswerModel.query().findByIds(ids);
+
+		return items.map((item) =>
+			QuizAnswerEntity.initialize({
+				createdAt: item.createdAt,
+				id: item.id,
+				label: item.label,
+				questionId: item.questionId,
+				updatedAt: item.updatedAt,
+				value: item.value,
+			}),
+		);
+	}
+
+	public async getCategoriezedAnswer(id: number): Promise<{
+		answerId: number;
+		categoryId: number;
+		value: number;
```

Comment:
I think you should create type for this

Reply:
I need to return object like that in this getCategorizedAnswer method only. So, type won't be reused. I believe I can omit creating separate type in that case